### ISSUE TITLE
Jaxb 2

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ankit4u3/jtracert#6]
```
What steps will reproduce the problem?
1. Using jtracert in J2EE apps which uses JAXB 2
2.
3.

What is the expected output? What do you see instead?
I see runtime exception. Execution of application fails:

com.sun.xml.bind.v2.runtime.IllegalAnnotationsException: 5 counts of 
IllegalAnnotationExceptions
Property jTracertObjectCompanion is present but not specified in 
@XmlType.propOrder
    this problem is related to the following location:
        at public 
com.google.code.jtracert.model.JTracertObjectCompanion 
sk.npvr.jaxb.smv.GetCustomerInfoInputType.jTracertObjectCompanion
        at sk.npvr.jaxb.smv.GetCustomerInfoInputType
        at public javax.xml.bind.JAXBElement 
sk.npvr.jaxb.smv.ObjectFactory.createGetCustomerInfoInput(sk.npvr.jaxb.smv.
GetCustomerInfoInputType)
        at sk.npvr.jaxb.smv.ObjectFactory


What version of the product are you using? On what operating system?
0.0.3, Windows

Please provide any additional information below.



```

Original issue reported on code.google.com by `igor.banduric` on 4 Jan 2009 at 2:45

**Comment by GoogleCodeExporter** [comment]
```
jTracert adds synthetic field jTracertObjectCompanion to all instrumented 
classes.
This field is used for calculating something like system hash code of objects, 
even
if hashCode() method of instrumented class was overwriten.

This approach is incorrect - we should use 
java.lang.System.identityHashCode(Object
x) method instead.

To be fixed in version 0.0.4
```

Original comment by `Dmitry.Bedrin` on 6 Jan 2009 at 10:19
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
I've added new sample application 'jaxbApplication' to the integration tests.
It's now reproducible in revision r97. 
```

Original comment by `Dmitry.Bedrin` on 27 Jan 2009 at 2:47

**Comment by GoogleCodeExporter** [comment]
```
Fixed in SVN revision r98
```

Original comment by `Dmitry.Bedrin` on 27 Jan 2009 at 3:43
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in jTracert 0.0.4
```

Original comment by `Dmitry.Bedrin` on 28 Jan 2009 at 8:37

**Comment by GoogleCodeExporter** [comment]
Original comment by `Dmitry.Bedrin` on 6 Feb 2009 at 12:38
- Changed state: **Verified**
